test/prompt/invocation_layer/test_sagemaker.py

```diff
@@ -0,0 +1,158 @@
+        yield mock_prompt_handler
+
+
+@pytest.mark.unit
+def test_default_constructor(mock_auto_tokenizer, mock_boto3_session):
+    """
+    Test that the default constructor sets the correct values
+    """
+
+    layer = SageMakerInvocationLayer(
+        model_name_or_path="some_fake_model",
+        max_length=99,
+        aws_access_key_id="some_fake_id",
+        aws_secret_access_key="some_fake_key",
+        aws_session_token="some_fake_token",
+        profile_name="some_fake_profile",
+        region_name="fake_region",
+    )
+
+    # assert layer. == "some_fake_key"
+    assert layer.max_length == 99
+    assert layer.model_name_or_path == "some_fake_model"
+
+    # assert mocked boto3 client called exactly once
+    mock_boto3_session.assert_called_once()
+
+    # assert mocked boto3 client was called with the correct parameters
+    mock_boto3_session.assert_called_with(
+        aws_access_key_id="some_fake_id",
+        aws_secret_access_key="some_fake_key",
+        aws_session_token="some_fake_token",
+        profile_name="some_fake_profile",
+        region_name="fake_region",
+    )
+
+
+@pytest.mark.unit
+def test_constructor_with_model_kwargs(mock_auto_tokenizer, mock_boto3_session):
+    """
+    Test that model_kwargs are correctly set in the constructor
+    and that model_kwargs_rejected are correctly filtered out
+    """
+    model_kwargs = {"temperature": 0.7, "do_sample": True, "stream": True}
+    model_kwargs_rejected = {"fake_param": 0.7, "another_fake_param": 1}
+
+    layer = SageMakerInvocationLayer(model_name_or_path="some_fake_model", **model_kwargs, **model_kwargs_rejected)
+    assert "temperature" in layer.model_input_kwargs
+    assert "do_sample" in layer.model_input_kwargs
+    assert "fake_param" not in layer.model_input_kwargs
+    assert "another_fake_param" not in layer.model_input_kwargs
+
+
+@pytest.mark.unit
+def test_invoke_with_no_kwargs(mock_auto_tokenizer, mock_boto3_session):
+    """
+    Test that invoke raises an error if no prompt is provided
+    """
+    layer = SageMakerInvocationLayer(model_name_or_path="google/flan-t5-xxl")
+    with pytest.raises(ValueError) as e:
+        layer.invoke()
+        assert e.match("No prompt provided.")
+
+
+@pytest.mark.unit
+def test_invoke_with_stop_words(mock_auto_tokenizer, mock_boto3_session):
+    """
+    Test stop words are correctly passed to HTTP POST request
+    """
+    stop_words = ["but", "not", "bye"]
+    layer = SageMakerInvocationLayer(model_name_or_path="some_model", api_key="fake_key")
+    with patch("haystack.nodes.prompt.invocation_layer.SageMakerInvocationLayer._post") as mock_post:
+        # Mock the response, need to return a list of dicts
+        mock_post.return_value = MagicMock(text='[{"generated_text": "Hello"}]')
+
+        layer.invoke(prompt="Tell me hello", stop_words=stop_words)
+
+    assert mock_post.called
+
+
+@pytest.mark.unit
+def test_ensure_token_limit_positive_mock(mock_boto3_session):
```

Comment:
Let's rename the test so that the name better explains what is tested. The name should mention that no resizing is triggered. For the next test, `test_ensure_token_limit_negative_mock`, the name should mention that resizing is triggered. As the resizing can only delete tokens but not add them, why not call it truncating instead?
For example, `test_short_prompt_is_not_truncated` and `test_long_prompt_is_truncated`.